app/controllers/multifactor_auths_controller.rb

```diff
@@ -71,18 +121,58 @@ def seed_and_expire
     end
   end
 
+  def verify_session_expiration
+    return if session_active?
+
+    delete_mfa_level_update_session_variables
+    redirect_to edit_settings_path, flash: { error: t("multifactor_auths.session_expired") }
+  end
+
+  def delete_mfa_level_update_session_variables
+    session.delete(:level)
+    session.delete(:webauthn_authentication)
+    delete_mfa_expiry_session
+  end
+
   # rubocop:disable Rails/ActionControllerFlashBeforeRender
   def handle_new_level_param
-    case level_param
+    case session[:level]
     when "disabled"
       flash[:success] = t("multifactor_auths.destroy.success")
       current_user.disable_totp!
     when "ui_only"
       flash[:error] = t("multifactor_auths.ui_only_warning")
     else
-      flash[:error] = t(".success")
-      current_user.update!(mfa_level: level_param)
+      flash[:error] = t("multifactor_auths.update.success")
+      current_user.update!(mfa_level: session[:level])
     end
   end
   # rubocop:enable Rails/ActionControllerFlashBeforeRender
+
+  def setup_mfa_authentication
```

Comment:
nit: `setup_totp_authentication`. Leaving this as a nit since it can be done later with the other `setup_mfa_authentication` s in the other controllers.

As an aside, could this behaviour be extracted somehow in the future (a concern perhaps).